### PR TITLE
Schema dump script

### PR DESCRIPTION
**Pull request by alloy**
Making this PR now so we can have discussions about how to do things, but probably ways off from merging.

### CODE REVIEW

**Comment by alloy** [comment]
What would be a good time to dump the schema? Like, on every git commit (i.e. in a pre-commit hook) or maybe have CI do it or something?

**Comment by mzikherman** [comment]
Dont you mean

look-who-i-ran-into-last-sunday_o_91196

**Comment by dzucconi** [comment]
So why is it that we have to update this static file anyway if it can be queried?

**Comment by alloy** [comment]
@dzucconi `schema.graphql` and `schema.json` both can? My _assumptions_ were that you couldn’t normally get the former and you might now want to serve the latter for performance reasons.

But I’m not sure how much time it takes to generate when all code has already been loaded, on the CLI it takes ~5 sec to generate both.

**Comment by dzucconi** [comment]
Oh, sorry I misunderstood—you right

**Comment by alloy** [comment]
I’m not sure yet what a good way is to ensure that the dumped schema gets updated when any of you make any changes to it, but for now I’m thinking that I could also do the actual dumping in emission.

I do think it makes sense to add the functionality of creating a schema dump to metaphysics, though, so that’s the `package.json` addition and the dump script.

**Comment by alloy** [comment]
@dzucconi bump

**Comment by dzucconi** [comment]
I'm not entirely sure why we'd want or need to commit this? Couldn't emission simple run the introspection query and cache it?

**Comment by alloy** [comment]
@dzucconi That’s pretty much what I’m proposing now. So I would remove the serialised schemas from the PR but leave the script that does the serialisation and its dependency.

* The reason that would still have to go into metaphysics is that it requires metaphysics’ npm env to be able to perform the serialisation.
* The reason it needs to be done by a script is because afaik that’s the only way to get a dump of the schema in `graphql` format.

**Comment by dzucconi** [comment]
Alright—works for me if you peel off those commits and pass lint we can get this merged

**Comment by alloy** [comment]
@dzucconi Did done.

**Comment by alloy** [comment]
Wait… this is missing the actual script.

File: scripts/dump-schema.js [hunk 1]
```diff
@@ -0,0 +1,30 @@
+import fs from 'fs';
+import path from 'path';
+import schema from '../schema';
+import { graphql }  from 'graphql';
+import { introspectionQuery, printSchema } from 'graphql/utilities';
+
+const destination = process.argv[2];
+if (destination === undefined || !fs.existsSync(destination)) {
+  console.error('Usage: dump-schema.js /path/to/output/directory');
```
Comment (dzucconi) [inline scripts/dump-schema.js]: You could either use the `debug` module or simply disable lint for this line/rule

Comment (alloy) [inline scripts/dump-schema.js]: Yeah I just disabled it.

**Comment by dzucconi** [comment]
👍

**Pull request merged**